Add more smart pointers to GeoNotifier

abigailfox opened the pull request:
#### 2243143ec5cff20e6b4305a2fb56108779643edc
<pre>
Add more smart pointers to GeoNotifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=275278">https://bugs.webkit.org/show_bug.cgi?id=275278</a>
<a href="https://rdar.apple.com/129419913">rdar://129419913</a>

Reviewed by Charlie Wolfe.

* Source/WebCore/Modules/geolocation/GeoNotifier.cpp:
(WebCore::GeoNotifier::runSuccessCallback):
(WebCore::GeoNotifier::runErrorCallback):
(WebCore::GeoNotifier::timerFired):
* Source/WebCore/Modules/geolocation/GeoNotifier.h:
(WebCore::GeoNotifier::protectedSuccessCallback):

Canonical link: <a href="https://commits.webkit.org/280045@main">https://commits.webkit.org/280045@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3c760f87c0e1117109513d4c6a59213dc1b91749

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55594 "Failed to checkout and rebase branch from PR 29641") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34917 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58578 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/6024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57720 "Failed to checkout and rebase branch from PR 29641") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42539 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6223 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/6024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57623 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/32820 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/47913 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/25895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/29606 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/5242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4168 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/5509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60169 "Built successfully") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30748 "Failed to checkout and rebase branch from PR 29641") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/5636 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/52196 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/31833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/47983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/32914 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8197 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/31580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebCore/Modules/geolocation/GeoNotifier.cpp:
```diff
@@ -105,26 +106,28 @@ void GeoNotifier::timerFired()
     // disconnected and requests are cancelled.
-    if (m_fatalError) {
-        runErrorCallback(*m_fatalError);
+    RefPtr fatalError = m_fatalError;
```
You could put this one in the if to restrict the scope

charliewolfe commented on Source/WebCore/Modules/geolocation/GeoNotifier.cpp:
```diff
@@ -76,13 +76,14 @@ void GeoNotifier::runSuccessCallback(GeolocationPosition* position)
 {
-    if (m_errorCallback)
-        m_errorCallback->handleEvent(error);
+    RefPtr errorCallback = m_errorCallback;
```
This one too

abigailfox committed:
Add more smart pointers to GeoNotifier
https://bugs.webkit.org/show_bug.cgi?id=275278
rdar://129419913

Reviewed by Charlie Wolfe.

* Source/WebCore/Modules/geolocation/GeoNotifier.cpp:
(WebCore::GeoNotifier::runSuccessCallback):
(WebCore::GeoNotifier::runErrorCallback):
(WebCore::GeoNotifier::timerFired):
* Source/WebCore/Modules/geolocation/GeoNotifier.h:
(WebCore::GeoNotifier::protectedSuccessCallback):

Canonical link: https://commits.webkit.org/280045@main
